### PR TITLE
:sparkles: DOP-4624 now using mongodb-chatbot-ui component's dark mode

### PR DESCRIPTION
**Pull request by caesarbell**
### Stories/Links:

DOP-4624

### Current Behavior:

[Doc's Landing](https://www.mongodb.com/docs/)

[Docs Python Language](https://www.mongodb.com/docs/languages/python/)

### Staging Links:

**Enabled Dark Mode**

[Staged: Doc's Landing](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/dark-mode-enabled/master/landing/caesarbell/DOP-4624/index.html)

[Staged: Docs Python Language](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/dark-mode-enabled/master/landing/caesarbell/DOP-4624/languages/python/index.html)

**Disabled Dark Mode** (for testing regression)

[Staged: Doc's Landing](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/dark-mode-disabled/master/landing/caesarbell/DOP-4624/index.html)

[Staged: Docs Python Language](https://docs-mongodb-org-stg.s3.us-east-2.amazonaws.com/dark-mode-disabled/master/landing/caesarbell/DOP-4624/languages/python/index.html)

### Notes:

- Removes `ChatbotUI` from the Component Factory, as it is not driven by a directive anymore
- Removes some custom style overrides from snooty, to allow the Chatbot UI Component's dark mode to work as designed
- Uses dark mode prop from the new action bar
- Update the test for the chatbot to reflect it's new home

### README updates

- - [ ] This PR introduces changes that should be reflected in the README, and I have made those updates.
- - [x] This PR does not introduce changes that should be reflected in the README

### CODE REVIEW

**Review by mayaraman19 — COMMENTED** [review]
a couple initial comments:
- the action bar is quite tall, compared to the current behavior/figma:
image
- and the last 2 staging links route back to this PR!

**Comment by caesarbell** [comment]
> a couple initial comments:
> 
> * the action bar is quite tall, compared to the current behavior/figma:
> 
> image - and the last 2 staging links route back to this PR!

@mayaraman19 good call on the height. I will set the height to match the Figma designs, which seems to be set at `96px`. For the second question, I am not sure what you mean. 

> and the last 2 staging links route back to this PR!
When I click on the last two links they take me to the page with dark mode disabled.

**Comment by mayaraman19** [comment]
> When I click on the last two links they take me to the page with dark mode disabled.

They work now, maybe my github was just broken 🥶

**Comment by caesarbell** [comment]
> > When I click on the last two links they take me to the page with dark mode disabled.
> 
> They work now, maybe my github was just broken 🥶

Github scared me for a second.

**Review by mmeigs — COMMENTED** [review]
It looks really good in the standard screen size!! 

1. I do wonder if we should ask design/chatbot teams about the placeholders (both) text color... I'm no professional but the contrast seems a bit too close to be accessible.

2. But the main thing I'm seeing is the responsiveness of the chatbot throughout smaller screensizes. Attaching a couple pics below. As soon as the screen becomes smaller than 1290px wide, it begins to have a mind of its own!

Screenshot 2024-06-18 at 9 14 09 AM
Screenshot 2024-06-18 at 9 14 25 AM

File: src/components/ChatbotUi.js [hunk 1]
```diff
@@ -103,13 +101,21 @@ const ChatbotUi = ({ template }) => {
     <StyledChatBotUiContainer data-testid="chatbot-ui" template={template}>
       {/* We wrapped this in a Suspense. We can use this opportunity to render a loading state if we decided we want that */}
       <SuspenseHelper fallback={<Skeleton borderRadius={SKELETON_BORDER_RADIUS} height={48} />}>
-        <Chatbot maxInputCharacters={DEFAULT_MAX_INPUT} serverBaseUrl={CHATBOT_SERVER_BASE_URL}>
-          <DocsChatbot suggestedPrompts={defaultSuggestedPrompts} />
+        <Chatbot maxInputCharacters={DEFAULT_MAX_INPUT} serverBaseUrl={CHATBOT_SERVER_BASE_URL} darkMode={darkMode}>
+          <DocsChatbot suggestedPrompts={defaultSuggestedPrompts} darkMode={darkMode} />
```
Comment (seungpark) [inline src/components/ChatbotUi.js:105]: Do we still want to use the [DocsChatbot component](https://github.com/mongodb/chatbot/blob/mongodb-chatbot-ui-v0.7.2/packages/mongodb-chatbot-ui/src/DocsChatbot.tsx#L27)? this is default with the disclosure at the bottom, but it seems like this is not in the Figma. We can update this to use more singular elements ie. <InputbarTrigger> without a bottomContent ([docs here](https://github.com/mongodb/chatbot/tree/mongodb-chatbot-ui-v0.7.2/packages/mongodb-chatbot-ui#usage))

**Review by seungpark — COMMENTED** [review]
Wondering if we still need the legal disclosure. Also, will need to address the mobile. I'm thinking we should not allow wrap for text here, and do a single line but we can check with Design

**Comment by caesarbell** [comment]
@mmeigs thank you for your feedback

> I do wonder if we should ask design/chatbot teams about the placeholders (both) text color... I'm no professional but the contrast seems a bit too close to be accessible.

There was a thread around the search bar and action bar. I didn't mention the contrast of the placeholder, which I can bring up, but I figured I would add the [link](https://mongodb.slack.com/archives/C076Z47AFTL/p1718641181796239) for more context. 

> But the main thing I'm seeing is the responsiveness of the chatbot throughout smaller screen sizes. Attaching a couple pics below. As soon as the screen becomes smaller than 1290px wide, it begins to have a mind of its own!

Good catch on this. I will work on this. I did hear that the positioning of the action bar and search input might change.

**Comment by caesarbell** [comment]
@seungpark thank you for your feedback. 

> Wondering if we still need the legal disclosure. Also, will need to address the mobile. I'm thinking we should not allow wrap for text here, and do a single line but we can check with Design

As mentioned above, there was a conversation around the action bar and about the said text. We are keeping it for now, some updated designs are coming down the pipeline that will change the text layout and format. Also, the placement of things will change as well.

File: src/components/ChatbotUi.js (hunk 1 above)
Comment (caesarbell) [reply src/components/ChatbotUi.js:105]: @seungpark these are good points. We are going to have more designs coming down the pipeline that will provide more clarity one the final direction. But moving to the suggested component could be a great option.

Comment (caesarbell) [reply src/components/ChatbotUi.js:105]: Also, there seems to be some inconsistency with the designs in Figma for the chatbot UI. In the [homepage frame](https://www.figma.com/design/7cR6eWtFvKTnJIPL8ZsTaB/Content-Template-Projects?node-id=1347-75011&t=as4VuNHTR2yNrKcA-0) you can see the text is there, but in the [toggle frame](https://www.figma.com/design/7cR6eWtFvKTnJIPL8ZsTaB/Content-Template-Projects?node-id=2936-6854&t=as4VuNHTR2yNrKcA-0) you can see the text is not visible. There are some other inconsistencies as well, for example, the color of the icon.

**Review by mmeigs — APPROVED** [review]
Responsiveness looks great! Thank you for that change! 

Note: we do need to capture the ongoing need for work on the Action bar/Chatbot in a ticket, I think. Just so we don't forget it!

**Comment by caesarbell** [comment]
> Responsiveness looks great! Thank you for that change!
> 
> Note: we do need to capture the ongoing need for work on the Action bar/Chatbot in a ticket, I think. Just so we don't forget it!

Good call. The [ticket](https://jira.mongodb.org/browse/DOP-4770) has been created.

File: tests/unit/Chatbot.test.js [hunk 2]
```diff
@@ -1,14 +1,14 @@
...
 import { render, waitFor } from '@testing-library/react';
-import ComponentFactory from '../../src/components/ComponentFactory';
+import ActionBar from '../../src/components/ActionBar/ActionBar';
 
 jest.mock('../../src/hooks/use-site-metadata', () => ({
   useSiteMetadata: () => ({ reposDatabase: 'pool_test' }),
 }));
 
 describe('Chatbot Ui', () => {
   it('renders the chatbot through the ComponentFactor', async () => {
```
Comment (seungpark) [inline tests/unit/Chatbot.test.js:10]: minor. we can update this description

**Comment by mayaraman19** [comment]
looks great - as matt said, thank you for addressing the responsiveness, however i think the size of the action bar is back to being a little bit taller than it currently is in figma/prod:

image

vs

image
in prod

**Review by seungpark — APPROVED** [review]
this looks good. we can follow up with DOP-4770

**Comment by caesarbell** [comment]
> looks great - as matt said, thank you for addressing the responsiveness, however i think the size of the action bar is back to being a little bit taller than it currently is in figma/prod:
> 
> image
> vs
> 
> image in prod


> looks great - as matt said, thank you for addressing the responsiveness, however i think the size of the action bar is back to being a little bit taller than it currently is in figma/prod:
> 
> image
> vs
> 
> image in prod

@mayaraman19 thank you for bringing this up, this was removed to work with the responsiveness and should be okay for this ticket, we have a new ticket that we will work on when the final designs are in. The complete placement might change. Please let me know if this satisfies your feedback.

**Review by mayaraman19 — APPROVED** [review]

**Pull request merged**